src/odm_sharing/private/cons.py

```diff
@@ -271,16 +314,22 @@ def connect(paths: List[str], tables: Set[str] = set()
     # as 0/1, which we'll have to convert back (using previously detected bool
     # columns) to 'FALSE'/'TRUE' before returning the data to the user. This
     # happens in `odm_sharing.sharing.get_data`.
-    _check_datasources(paths)
+    if not data_sources:
+        raise DataSourceError('no data source')
     try:
-        path = paths[0]
-        is_csv = path.endswith('.csv')
-        if not is_csv and len(paths) > 1:
-            logging.warning('ignoring additional inputs (for CSV only)')
-
-        if is_csv:
-            return _connect_csv(paths)
-        elif path.endswith('.xlsx'):
+        if _detect_csv_input(data_sources):
+            csv_data_sources = cast(CsvDataSourceList, data_sources)
```

Comment:
Couldn't `data_sources` be a string? in which case casting it to CsvDataSourceList would be incorrect.

Reply:
good catch. fixed.